src/logic/DataUploader.ts

```diff
@@ -0,0 +1,31 @@
+const MINUTE = 1000 * 60;
+const INTERVAL_DURATION = 1 * MINUTE;
+
+export default class ServerLogic{
+    static interval: number = -1;
+    static temporaryData: any;
```

Comment:
this might need to be an array of data readings, like `storedReadings`. You can then push incoming `reading`to it and whenever your `logic` function is called, just upload all your `storedReadings` to it. When it is successful, you can reset it

Reply:
actually this object is directly comes from dummy generator or from serial communication, and it contain the array of readings (I presumed from dummy data). so every time new reading are added to graph the same will added here in this object.